client/components/jetpack/business-at-switch/index.tsx

```diff
@@ -0,0 +1,99 @@
+/**
+ * External dependencies
+ */
+import React, { FC, ReactElement, useState, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import { isEmpty } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Main from 'components/main';
+import FormattedHeader from 'components/formatted-header';
+import QueryEligibility from 'components/data/query-atat-eligibility';
+import WPCOMBusinessAT from 'components/jetpack/wpcom-business-at';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import {
+	getAutomatedTransfer,
+	isEligibleForAutomatedTransfer,
+} from 'state/automated-transfer/selectors';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+type Props = {
+	context: PageJS.Context;
+};
+
+const Placeholder = () => (
+	<>
+		<FormattedHeader
+			id="business-at-switch"
+			className="business-at-switch placeholder__header"
+			headerText="Loading..."
+			align="left"
+		/>
+
+		<div className="business-at-switch placeholder__primary-promo"></div>
+	</>
+);
+
+/**
+ * This component selects the right view for Business plan sites that
+ * are not Atomic. If we don't know if the site is eligible for Automated
+ * Transfer, it renders a placeholder and an instance of <QueryEligibility />.
+ * Once we have an answer, we either display the Automated Transfer view or
+ * we display whatever comes next in the middleware stack.
+ */
+const BusinessATSwitch: FC< Props > = ( { context } ): ReactElement => {
```

Comment:
I think it would be cleaner to do this instead of `wpcomUpsellController`, and let the `WPCOMBusinessAT` component check eligibility since that isn't a binary state.

Reply:
I think there is still value in having controlling whether we should display `WPCOMBusinessAT` in a parent component. If there is no need for AT, we skip the `WPCOMBusinessAT` component entirely. I made a small update to pass down to `WPCOMBusinessAT` the status of the Automated Transfer so we can deal with it inside the component.

Let me know what you think. If you still think we should follow your path, I'm 100% ok with that too.

Reply:
Nah, this path works. Thanks for updating!